kernel/kernel-api/src/test/scala/io/delta/kernel/internal/util/SchemaIterableSuite.scala

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.util
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ArrayBuffer
+
+import io.delta.kernel.internal.types.DataTypeJsonSerDe
+import io.delta.kernel.types._
+
+import org.scalatest.funsuite.AnyFunSuite
+
+class SchemaIterableSuite extends AnyFunSuite {
+  test("depth first traversal works with deeply nested types") {
+    val schema: StructType = getDeeplyNestedSchema
+
+    val iterable = new SchemaIterable(schema)
+
+    // Track the path stack during traversal
+    val fieldInfo = iterable.asScala.map {
+      element =>
+        (
+          element.getNamePath(),
+          element.getPathFromNearestStructFieldAncestor(
+            element.getNearestStructFieldAncestor.getName),
+          element.getPathFromNearestStructFieldAncestor(""),
```

Comment:
Are these common cases that we anticipate our code having?

Would adding some helper functions make these easier to understand? (And, more succinct?)

Reply:
Possibly, I think we can decide what makes sense to extend with helper methods as we get more use out of the iterable api.